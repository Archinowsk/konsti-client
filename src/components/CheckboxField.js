/* @flow */
import React from 'react'
import { useTranslation } from 'react-i18next'
import type { FieldProps } from 'redux-form'
import config from 'config'

const CheckboxField = (field: FieldProps) => {
  const { t } = useTranslation()

  if (!field) return <p />

  const { name } = field.input
  const { touched, error } = field.meta

  const registerDescriptionLink = `${config.apiServerURL}/rekisteriseloste.txt`

  return (
    <div className='form-row'>
      <div className='form-field'>
        <div className='form-input-wrapper' id={`${name}-wrapper`}>
          <input
            className='form-input checkbox'
            {...field.input}
            placeholder={t(name)}
            id={name}
            // $FlowFixMe: Cannot get field.type because property type is missing in FieldProps [1].
            type={field.type}
            label={t(name)}
          />
        </div>

        <label htmlFor={name}>
          {t('iAgree')}{' '}
          <a
            href={registerDescriptionLink}
            target='_blank'
            rel='noopener noreferrer'
          >
            {t('registerDescription')}
          </a>
        </label>
      </div>

      {touched && error && <div className='form-field-error'>{t(error)}</div>}
    </div>
  )
}

export default CheckboxField
