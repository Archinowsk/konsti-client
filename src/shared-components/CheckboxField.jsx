import React from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import config from '../../config';

const FormField = props => {
  const { input, label = '', type, meta: { touched, error }, t } = props;

  const registerDescriptionLink = `${config.apiServerURL}/rekisteriseloste.txt`;

  return (
    <div className="form-row">
      <div className="form-field">
        <div className="form-input-wrapper" id={`${input.name}-wrapper`}>
          <input
            className="form-input"
            {...input}
            placeholder={label}
            type={type}
            id={input.name}
          />
        </div>

        <label htmlFor={label}>
          {t('agreeTerms')}{' '}
          <a href={registerDescriptionLink} target="_blank">
            {t('registerDescription')}
          </a>
        </label>
      </div>

      {touched &&
        error &&
        <div className="form-field-error">
          {error}
        </div>}
    </div>
  );
};

FormField.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string, // eslint-disable-line react/require-default-props
  type: PropTypes.string.isRequired,
  meta: PropTypes.object.isRequired,
  t: PropTypes.func.isRequired,
};

export default translate()(FormField);
