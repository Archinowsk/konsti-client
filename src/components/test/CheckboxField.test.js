/* @flow */
import React from 'react'
import { shallow } from 'enzyme'
import CheckboxField from '../CheckboxField'
import type { Props } from '../CheckboxField'

describe('CheckboxField', () => {
  it('should render correctly', () => {
    const props: Props = {
      input: null,
      label: '',
      type: '',
      meta: { touched: false, error: null },
    }
    const component = shallow(<CheckboxField {...props} />)
    expect(component).toMatchSnapshot()
  })
})
