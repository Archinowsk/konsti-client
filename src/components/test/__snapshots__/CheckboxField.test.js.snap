// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxField should render correctly 1`] = `
<div
  className="form-row"
>
  <div
    className="form-field"
  >
    <div
      className="form-input-wrapper"
      id="password-wrapper"
    >
      <input
        className="form-input checkbox"
        id="password"
        label="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="password"
        value=""
      />
    </div>
    <label
      htmlFor="password"
    >
      iAgree
       
      <a
        href="http://localhost:3000/rekisteriseloste.txt"
        rel="noopener noreferrer"
        target="_blank"
      >
        registerDescription
      </a>
    </label>
  </div>
  <div
    className="form-field-error"
  >
    validation.required
  </div>
</div>
`;
